File: sys/src/geometry/point2.cc
#include "point2.h"

namespace gtsam {

std::unique_ptr<Point2> default_point2() { return std::make_unique<Point2>(); }

std::unique_ptr<Point2> new_point2(double x, double y) {
  return std::make_unique<Point2>(x, y);
}


} // namespace gtsam

File: src/geometry/point2.rs
use cxx::UniquePtr;

pub struct Point2 {
    pub(crate) inner: UniquePtr<::sys::Point2>,
}

impl Default for Point2 {
    fn default() -> Self {
        Self {
            inner: ::sys::default_point2(),
        }
    }
}

impl From<::nalgebra::Vector2<f64>> for Point2 {
    fn from(value: ::nalgebra::Vector2<f64>) -> Self {
        Self::new(value.x, value.y)
    }
}

impl Point2 {
    pub fn new(x: f64, y: f64) -> Self {
        Self {
            inner: ::sys::new_point2(x, y),
        }
    }
}
